server/services/correlation/recurrence_agent.py

```
@@ -0,0 +1,615 @@
+    finally:
+        try:
+            weaviate_client.close()
+        except Exception:
+            logger.exception("%s Failed to close weaviate client", _LOG_PREFIX)
+        try:
+            postgres_client.close()
+        except Exception:
+            logger.exception("%s Failed to close postgres client", _LOG_PREFIX)
+
+
+def run_recurrence_check(
```

Ruff (0.16.2)

[warning] Too many return statements (7 > 6)

(PLR0911)